Update vmware_vmkernel to support Distributed Virtual Switches

vmwjoseph opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Update vmware_vmkernel and vmware_vmkernel_facts to support Distributed Virtual Switches.

- I renamed `vswitch_name` to a more generic `switch_name` which matches other vmware ansible modules.
- Add new required parameter `device_name` so that devices can be looked up by name instead of assuming it is the only device on a portgroup.
- Added new optional parameter `is_dvs_switch` which will determine whether vmkernel device is being added to a VSS or DVS.
- Removed parameter `vlan_id` which was not used

<!--- If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request


##### COMPONENT NAME
<!--- Name of the module, plugin, task or feature -->
- vmware_vmkernel
- vmware_vmkernel_facts

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.6.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/opt/lib/ansible/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Sep 23 2017, 22:06:14) [GCC 7.2.0]
```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->

<!--- Paste verbatim command output below, e.g. before and after your change -->

# vmware_vmkernel After change (new functionality, DVS did not work at all previously)
```
## First pass, this does most of the work, but only adds vmnic1 as an uplink for DVS
root@cd250dd6f77e:/opt/nova/vsphere# `ansible-playbook playbooks/vcenter_config_dvs.yml -i ../core/cicd-dev/novaclean-01 --ask-vault-pass --extra-vars='vcenter_hostname=10.193.18.128 esx_hostgroup=esxi-hosts vcenter_dvs_esxi_vmnic=vmnic1' --module-path ../ansible-modules/nioc/ --module-path ../ansible-modules/vmk/`
```text
Vault password: 

...

TASK [../esxi/roles/config-vmkernel-int : Manage VMkernel interfaces] ***********************************************************************************************************************************************************************************************
ok: [10.193.0.134 -> localhost] => (item={u'ft': False, u'name': u'vmk0', u'portgroup': u'mgmt', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'vlan_id': 0, u'mgmt': True, u'mtu': 1500, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})
ok: [10.193.17.67 -> localhost] => (item={u'ft': False, u'name': u'vmk0', u'portgroup': u'mgmt', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'vlan_id': 0, u'mgmt': True, u'mtu': 1500, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})
ok: [10.193.2.215 -> localhost] => (item={u'ft': False, u'name': u'vmk0', u'portgroup': u'mgmt', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'vlan_id': 0, u'mgmt': True, u'mtu': 1500, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})
ok: [10.193.0.223 -> localhost] => (item={u'ft': False, u'name': u'vmk0', u'portgroup': u'mgmt', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'vlan_id': 0, u'mgmt': True, u'mtu': 1500, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})
changed: [10.193.0.134 -> localhost] => (item={u'ft': False, u'name': u'vmk1', u'portgroup': u'vmotion', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': True})
changed: [10.193.0.223 -> localhost] => (item={u'ft': False, u'name': u'vmk1', u'portgroup': u'vmotion', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': True})
changed: [10.193.17.67 -> localhost] => (item={u'ft': False, u'name': u'vmk1', u'portgroup': u'vmotion', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': True})
changed: [10.193.2.215 -> localhost] => (item={u'ft': False, u'name': u'vmk1', u'portgroup': u'vmotion', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': False, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': True})
changed: [10.193.0.134 -> localhost] => (item={u'ft': False, u'name': u'vmk2', u'portgroup': u'vsan', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': True, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})
changed: [10.193.17.67 -> localhost] => (item={u'ft': False, u'name': u'vmk2', u'portgroup': u'vsan', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': True, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})
changed: [10.193.0.223 -> localhost] => (item={u'ft': False, u'name': u'vmk2', u'portgroup': u'vsan', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': True, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})
changed: [10.193.2.215 -> localhost] => (item={u'ft': False, u'name': u'vmk2', u'portgroup': u'vsan', u'network': {u'type': u'dhcp', u'netmask': None, u'ip_address': None}, u'state': u'present', u'mgmt': False, u'mtu': 9000, u'vsan': True, u'switch_name': u'nova_dvs', u'dvs': True, u'vmotion': False})

...
```


# vmware_vmkernel_facts Before change
```
root@ff83ee75f453:/opt/nova/vsphere# ansible-playbook playbooks/vmkernel-facts.yml -i ../core/cicd-dev/novaclean-01 --ask-vault-pass --extra-vars='vcenter_hostname=10.193.18.128 esx_hostgroup=10.193.17.67'

Vault password: 

PLAY [10.193.17.67] *****************************************************************************************************************************************************************************************

TASK [../esxi/roles/vmkernel-facts : Gather VMKernel facts about all ESXi Host in given Cluster] ************************************************************************************************************
ok: [10.193.17.67]

TASK [../esxi/roles/vmkernel-facts : debug] *****************************************************************************************************************************************************************
ok: [10.193.17.67] => {
    "msg": {
        "changed": false, 
        "failed": false, 
        "host_vmk_facts": {
            "10.193.17.67": [
                {
                    "device": "vmk0", 
                    "dhcp": true, 
                    "enable_ft": false, 
                    "enable_management": true, 
                    "enable_vmotion": false, 
                    "enable_vsan": false, 
                    "ipv4_address": "10.193.17.67", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk0", 
                    "mac": "02:00:4a:76:3e:66", 
                    "mtu": 1500, 
                    "portgroup": "", 
                    "stack": "defaultTcpipStack"
                }, 
                {
                    "device": "vmk1", 
                    "dhcp": true, 
                    "enable_ft": false, 
                    "enable_management": false, 
                    "enable_vmotion": true, 
                    "enable_vsan": false, 
                    "ipv4_address": "10.193.8.145", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk1", 
                    "mac": "00:50:56:68:0c:b5", 
                    "mtu": 9000, 
                    "portgroup": "", 
                    "stack": "defaultTcpipStack"
                }, 
                {
                    "device": "vmk2", 
                    "dhcp": true, 
                    "enable_ft": false, 
                    "enable_management": false, 
                    "enable_vmotion": false, 
                    "enable_vsan": true, 
                    "ipv4_address": "10.193.9.233", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk2", 
                    "mac": "00:50:56:6f:50:0e", 
                    "mtu": 9000, 
                    "portgroup": "", 
                    "stack": "defaultTcpipStack"
                }, 
                {
                    "device": "vmk3", 
                    "dhcp": true, 
                    "enable_ft": true, 
                    "enable_management": false, 
                    "enable_vmotion": false, 
                    "enable_vsan": false, 
                    "ipv4_address": "10.193.27.17", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk3", 
                    "mac": "00:50:56:63:e8:d1", 
                    "mtu": 1500, 
                    "portgroup": "ft_vss", 
                    "stack": "defaultTcpipStack"
                }
            ]
        }
    }
}

PLAY RECAP **************************************************************************************************************************************************************************************************
10.193.17.67               : ok=2    changed=0    unreachable=0    failed=0   
```
# vmware_vmkernel_facts After change

```
root@ff83ee75f453:/opt/nova/vsphere# ansible-playbook playbooks/vmkernel-facts.yml -i ../core/cicd-dev/novaclean-01 --ask-vault-pass --extra-vars='vcenter_hostname=10.193.18.128 esx_hostgroup=10.193.17.67' --module-path=/opt/nova/ansible-modules/vmk/

Vault password: 

PLAY [10.193.17.67] *****************************************************************************************************************************************************************************************

TASK [../esxi/roles/vmkernel-facts : Gather VMKernel facts about all ESXi Host in given Cluster] ************************************************************************************************************
ok: [10.193.17.67]

TASK [../esxi/roles/vmkernel-facts : debug] *****************************************************************************************************************************************************************
ok: [10.193.17.67] => {
    "msg": {
        "changed": false, 
        "failed": false, 
        "host_vmk_facts": {
            "10.193.17.67": [
                {
                    "device": "vmk0", 
                    "dhcp": true, 
                    "dvs_name": "nova_dvs", 
                    "enable_ft": false, 
                    "enable_management": true, 
                    "enable_vmotion": false, 
                    "enable_vsan": false, 
                    "ipv4_address": "10.193.17.67", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk0", 
                    "mac": "02:00:4a:76:3e:66", 
                    "mtu": 1500, 
                    "portgroup": "mgmt", 
                    "stack": "defaultTcpipStack"
                }, 
                {
                    "device": "vmk1", 
                    "dhcp": true, 
                    "dvs_name": "nova_dvs", 
                    "enable_ft": false, 
                    "enable_management": false, 
                    "enable_vmotion": true, 
                    "enable_vsan": false, 
                    "ipv4_address": "10.193.8.145", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk1", 
                    "mac": "00:50:56:68:0c:b5", 
                    "mtu": 9000, 
                    "portgroup": "vmotion", 
                    "stack": "defaultTcpipStack"
                }, 
                {
                    "device": "vmk2", 
                    "dhcp": true, 
                    "dvs_name": "nova_dvs", 
                    "enable_ft": false, 
                    "enable_management": false, 
                    "enable_vmotion": false, 
                    "enable_vsan": true, 
                    "ipv4_address": "10.193.9.233", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk2", 
                    "mac": "00:50:56:6f:50:0e", 
                    "mtu": 9000, 
                    "portgroup": "vsan", 
                    "stack": "defaultTcpipStack"
                }, 
                {
                    "device": "vmk3", 
                    "dhcp": true, 
                    "enable_ft": true, 
                    "enable_management": false, 
                    "enable_vmotion": false, 
                    "enable_vsan": false, 
                    "ipv4_address": "10.193.27.17", 
                    "ipv4_subnet_mask": "255.255.224.0", 
                    "key": "key-vim.host.VirtualNic-vmk3", 
                    "mac": "00:50:56:63:e8:d1", 
                    "mtu": 1500, 
                    "portgroup": "ft_vss", 
                    "stack": "defaultTcpipStack"
                }
            ]
        }
    }
}

PLAY RECAP **************************************************************************************************************************************************************************************************
10.193.17.67               : ok=2    changed=0    unreachable=0    failed=0   
```

Commit:
Update vmware_vmkernel to support Distributed Virtual Switches

Commit:
Update vmware_vmkernel to support Distributed Virtual Switches

Commit:
Fix ansible-test sanity failures

Commit:
Fix ANSIBLE_METADATA.metadata_version

Akasurde commented on lib/ansible/modules/cloud/vmware/vmware_vmkernel.py:
```diff
@@ -4,6 +4,8 @@
 # Copyright: (c) 2015, Joseph Callen <jcallen () csc.com>
 # Copyright: (c) 2017-18, Ansible Project
 # Copyright: (c) 2017-18, Abhijeet Kasurde <akasurde@redhat.com>
+# Copyright 2018 VMware, Inc.
```
```# Copyright: (c) 2018, VMware, Inc.```

Akasurde commented on lib/ansible/modules/cloud/vmware/vmware_vmkernel.py:
```diff
@@ -27,18 +29,32 @@
     - PyVmomi
 options:
-    vswitch_name:
+    switch_name:
```
rather than removing parameter, add `alias` as 

```
aliases: ['switch_name']
```

Akasurde commented on lib/ansible/modules/cloud/vmware/vmware_vmkernel.py:
```diff
@@ -27,18 +29,32 @@
+        version_added: 2.7
+    is_dvs_switch:
+        description:
+            - Whether or not C(switch_name) is a DVS
```
period at the end of statement.

Akasurde commented on lib/ansible/modules/cloud/vmware/vmware_vmkernel.py:
```diff
@@ -116,44 +125,56 @@
-      username: '{{ esxi_username }}'
-      password: '{{ esxi_password }}'
-      vswitch_name: vSwitch0
+      hostname: 192.168.127.9
```
Use Jinja variable instead of IPs.

Akasurde commented on lib/ansible/modules/cloud/vmware/vmware_vmkernel.py:
```diff
@@ -116,44 +125,56 @@
       portgroup_name: PG_0002
-      vlan_id: '{{ vlan_id }}'
       state: absent
-   delegate_to: localhost
```
`delegate_to` is required.

Akasurde commented on lib/ansible/modules/cloud/vmware/vmware_vmkernel.py:
```diff
@@ -198,7 +220,14 @@ def __init__(self, module):
+                module.fail_json(msg="Portgroup name %s not found" % self.port_group_name)
+        else:
+            self.dv_switch = find_dvs_by_name(self.content, self.switch_name)
+            self.port_group_obj = find_dvspg_by_name(self.dv_switch, self.port_group_name)
```
Add check to see if we found right port_group_obj under `dvswitch` or not.

Akasurde commented on lib/ansible/modules/cloud/vmware/vmware_vmkernel_facts.py:
```diff
@@ -57,7 +60,6 @@
     username: '{{ vcenter_username }}'
     password: '{{ vcenter_password }}'
     esxi_hostname: '{{ esxi_hostname }}'
-  delegate_to: localhost
```
delegate_to is required in example.

maxamillion committed:
Improve dnf group output for clarity

Add note about group removal bug upstream dnf

Signed-off-by: Adam Miller <admiller@redhat.com>

mattclay committed:
Flush output from timing script.

nitzmahone committed:
initial commit of azure_rm inventory plugin (#44944)

* crusty refactor of azure_rm to support auth from non-modules

bonzi316 committed:
Fix maven_artifact md5 checksum check with archiva repository (#43250)

ganeshrn committed:
Fix netconf_config module default_operation issue (#44958)

* Pass parameters as dict to edit_config api
  as either dict or args can be passed over
  jsonrpc 2.0 and not combination of args and
  kwargs

Commit:
maven_artifact: download with shutil.copyfileobj to a temp file (#44956)

* maven_artifact: replace homemade chunked download with shutil.copyfileobj

* maven_artifact: copy to temp file first

rambleraptor committed:
removing libcloud secrets support for legacy gcp modules (#44932)

* removing libcloud secrets support for legacy gcp modules

* test fixes

gdpak committed:
cisco firepower : Make API endpoints configurable via hostvars (#44952)

* httpapi host vars

* Make configurable end-points for firepower

* pep8 fix

bcoca committed:
clarify priority setting (#44865)

* clarify priority setting

* updated as per fb

samccann committed:
fixed grammar and spelling issues (#44931)

* fixed grammar and spelling issues

* fixed autocorrect mistake and sanity test errors

jamezpolley committed:
Improve clarity of precedence when command-line parameters are used. (#39059)

* Improve clarity of precedence when command-line parameters are used.

* Add command-line values into the precedence list.
* Several sample config snippets were included without any explanation
  of how those snippets would be processed. Added descriptions so that
  the reader can understand what each snippet will (or won't) accomplish.

* Don't focus on inventory as much

Expand on the fact that it's the fact that a variable is set that
matters, not the source of the variable.

ganeshrn committed:
Fix iosxr_config integration test failure (#44969)

*  Check for commands presence in case running config
   is used as base config for diff
*  Correct the fixtures

bcoca committed:
added more reasons to avoid dot notation (#43724)

ganeshrn committed:
Fix cli_command multiple prompt issue (#44922)

* Add check in network_cli to handle all prompts

* Add check_all flag to mandatory handle all the command prompt
  in prompts list. By default if any one prompt is handled
  remaining prompts are ignored.

* Fix cli_command multiple prompt issue

* If multiple prompt and answers are given as input network_cli
  handles only the first prompt that matched by default
* If a command execution results in muliple prompt the fix
  add support to set a boolean option C(check_all) to indicate
  network_cli to wait till all the prompts and answers are processed.

* Update cli_command

* Update api doc

* Fix unit test failure

* Fix CI failure

* Update network_cli

* Fix review comment

Commit:
Lenovo image refactor (#44953)

* Refactoring code to adhere to persistence connection.

* Update cnos_rollback.py

* Updating license for the refactored method

* Update cnos_rollback.py

* Removing the BSD License as suggested by Legal

* Adding cnos_image after testing on sftp. Test code is also added

* Removing debug comments

* adding space

Spredzy committed:
Vultr: Create the Vultr Guide (#44515)

cedlerouge committed:
Fix autostart description 'network' instead of 'storage pool' (#28515)

Commit:
maven_artifact: enforce required params (fixes #44823) (#44967)

samdoran committed:
Change Mac OSX to macOS in installation docs (#44971)

* Change Mac OSX to macOS in docs

* Lowercase via

remyleone committed:
Add an user facts module for Online (#44709)

abadger committed:
Reset changelogs for the 2.8.0 development cycle

abadger committed:
Update ansible version number to 2.8.0.dev0

abadger committed:
Also remove the .changes.yaml file

dagwieers committed:
Add elapsed return value to select modules (#37969)

* Add elapsed return value to select modules

It can be quite useful to know exactly how much time has elapsed
downloading/waiting. This improves existing modules or updates
documentation.

* Ensure elapsed is always returned

* Added changelog fragment

sivel committed:
Stub out modules scheduled for 2.8 removal (#44985)

* Remove modules scheduled for 2.8

* Add changelog and porting guide

* Skip docs test on removed modules

* Don't link to removed modules

Commit:
added missing public ip address alias

jonjitsu committed:
Added default value for when CloudFrontOriginAccessIdentityList is (#44984)

missing.

farrellit committed:
Update lineinfile.py (#44976)

* Update lineinfile.py

The document should call out in the regexp section what regexp does in non-match scenario, not just leave it for the reader to find it under `insertbefore` and `insertafter`.  
+label: docsite_pr

* Update lineinfile.py

fixing for lint

* Use multiple entries

mattclay committed:
Fix PEP8 issue.

felixfontein committed:
Fixing typo in acme_certificate docs. (#45082)

rndmh3ro committed:
remove myself from bot-notifications (#45087)

Commit:
added postgresql management client version (#45079)

Commit:
merging postgresql server and database facts (#45061)

yungezz committed:
improve vmss test (#44892)

* improve vmss test

* fix lint

* fix lint

* fix lint

* fix  test bug

vpopescu committed:
fixed boolean value for zone (#41974)

yungezz committed:
fix blob in non-public azure cloud (#44802)

* fix blob in non-public azure cloud

* fix  lint

* fix typo

remyleone committed:
Add a dynamic inventory plugin for Online (#44720)

Commit:
adding container instance facts (#43326)

Commit:
merge mysql server and database tests (#45060)

remyleone committed:
Add licence and boilerplate to Online module_utils

dagwieers committed:
aci_switch_policy_vpc_protection_group: Fix idempotency (#45091)

dagwieers committed:
aci_interface_policy_leaf_policy_group: Fix filtering by lagT (#45088)

mattclay committed:
Update test constraints. (#44975)

- Limit virtualenv version on Python 2.6.
- Limit pyopenssl version on Python 2.6.
- Pin pyfmg version to keep unit tests passing.

Madhura-CSI committed:
New module for azure custom image facts (#42102)

* new module for azure custom image facts

* added facts in aliases, corrected assertion

* simplified managed disk id struct

* updated ansible version

Commit:
added mysql management client to remove api version warning (#45103)

Commit:
vm size idempotence (#45108)

Commit:
Virtual machine facts - retrieving instance view (#45128)

Commit:
consolidate sql tests into one (#45064)

skg-net committed:
Docs updated and review comments addressed (#45133)

abadger committed:
Add a table of contents to the generated changelog (#45046)

linnil1 committed:
Fix sshkeyfilename is None not str (#44893)

ganeshrn committed:
Fix netconf module and plugin issues (#45140)

* Fix netconf netconf issues

*  Identifier is optional for get_schema api
*  Fix dispatch api mandatory argument check
*  Add save option handling to copy config from target datastore
   to startup datastore if supported
*  Validate config in check-mode or if validate option set to true

* Copy config if check-mode is not enabled

Commit:
adding tags to container instance (#45142)

Commit:
added missing resource_group description in return value (#45143)

trishnaguha committed:
fix Python 2.6 regex bug terminal plugin nxos, iosxr (#45135)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

Akasurde committed:
Documentation for several VMware modules (#45153)

* Added return for vmware_host_config_facts
* Added return for vmware_host_service_facts
* Correct typo in vmware_portgroup_facts

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Handle special service where package is not available (#45155)

There are several services e.g. vmware-fdm, which does not have package name and
package description which will raise a error if queried for.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ryansb committed:
Add rds_instance to module_defaults groups

Akasurde committed:
Add support for Devuan (#45148)

Devuan is part of Debian family. This fix adds support for Devuan OS.

Fixes: #45047

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

characteristic committed:
Correct Jinja2 plugin math filter symmetric_difference() to not repeatedly (#45093)

build its intersection set and unnecessarily unique the final result.

The prior use of the intersect() function within the list comprehension
conditional leads to the function being called for every value in the input
list being processed, not efficient.  When the input lists a,b are large,
the Ansible run time and resource utilization wildly increases generally
never completing the operation.

Unique of the intersection result is unnecessary as the source list union()
is already unique.

mcameron committed:
Fixes ec2.py assume_role. (#37357)

* Fixes ec2.py assume_role

Previously when connect_to_aws was called it updated the credentials in place.  "connect_to_aws" is called multiple times: on the second run it tries to assume the same role it is already using, and potentially failing depending on your iam policies.

joren485 committed:
Fix dead code in files module (#43479)

* Use pwd.getpwnam instead of non-existent pwd.getpwname

* Add specific exceptions to pwd and grp calls

afair committed:
Returns reason on pkg update error (#44982)

Previously, it returned no additional information:

     "msg": "Could not update catalogue"

Now it passes that reason with the error message:

    {"changed": false, "msg": "Could not update catalogue [77]:  pkg: Insufficient privileges to update the repository catalogue.\n"}

samccann committed:
fixed formatting (#45164)

(cherry picked from commit 58ca82fa1c59d3a62509db99c4bbff6eb08829ab)

Protista committed:
Add definitive to valid_simple_controls in pamd module (#44601)

* Add definitive to valid_simple_controls in pamd module

* Add changelog fragment for issue 44278

* Adjusted module name to not include path or extension

dseeley-sky committed:
ec2_vpc_route_table - allow routes to be created if the CIDR already exists but its 'Origin' is 'EnableVgwRoutePropagation' (which cannot be replaced). (#43417)

Fixes #43415

dagwieers committed:
Improve GitHub templates (#44455)

* Improve GitHub templates

The existing templates are basically a copy for each type of issue (bug,
feature, doc) whereas we can make the template more specific to the task
at hand.

This PR includes:
- More specific sections depending on the type of issue
- More concise and direct instructions (NO NEED FOR FRUSTRATION CAPS!)
- Single-line comments (no need to guide people to avoid wrong edits)
- Improved paste area
- Better headings/titles
- No 'Summary' title for PRs (so commit information is placed correctly)

This PR would require some improvements to Ansibot.

* Remove controversial Ansibot stuff

Make it easier to get the first batch merged.

* Restore ISSUE_TEMPLATE.md

* Incorporate reviewer suggestions

* Make OS / ENVIRONMENT doc-related

* Rephrase Ansibot-related message

mattclay committed:
Skip ipaddr test that fails on Python 3.7+.

mattclay committed:
Switch default test container to quay.io.

mattclay committed:
Fix ansible-test virtualenv use in import test.

ryansb committed:
Codename: How Many More Times

felixfontein committed:
Refactoring GET request handling. (#45051)

siwyd committed:
ec2 inventory: python 3.7 compatibility (#43716)

ec2 inventory script was throwing errors when using Python 3.7:

TypeError: option values must be strings

This changes the None ConfigParser options to empty strings instead.

remyleone committed:
Add user_data/cloud-init support for scaleway_compute module (#42697)

mintsoft committed:
Subversion: Only fire changed when content has changed  (#41036)

* Previously when checking out a subdirectory, if changes are made in the
repository that aren't present in the subdirectory (i.e. in a different
branch or similiar) then the changes handler still kicks in even though
the content of this working copy hasn't necessarily changed. Now we parse
the output of svn update and fire if anything changes at all.

* Previously when checking out a subdirectory, if changes are made in the
repository that aren't present in the subdirectory (i.e. in a different
branch or similiar) then the changes handler still kicks in even though
the content of this working copy hasn't necessarily changed. Now we parse
the output of svn update and svn switch, firing if anything changes at all.

* Should not be executable

* ==None doesn't do what I want, is None does

BondAnthony committed:
DigitalOcean inventory deprecation warning (#45054)

Handle DeprecationWarning for ConfigParser in python3

Fixes #45053

Akasurde committed:
VMware: Return Disk UUID information (#45157)

This fix adds feature of disk uuid for given virtual machine disk.

Fixes: #45149

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
adding tags to keyvault (#45145)

Commit:
Consolidate keyvault tests (#45196)

amb1s1 committed:
Fixed an issue when expecting may be, but getting maybe (#44908)

Tested an install on a Nexus 5596 and was failing due that this model will send an output of "Another install procedure maybe in progress" when running the show install all while the install is running. This is how the module knows that an install is being processed. In my case, it will not match that conditional statement.

abirami-n committed:
enable_netconf_support_on_dellos10 (#44816)

bcoca committed:
fixed chlog path

ryansb committed:
[aws] Add suboptions to aws_application_scaling_policy (#43042)

* [aws] Add suboptions to aws_application_scaling_policy

Fixes #42917

* Remove any suboptions that are None

s-hertel committed:
elb_target_group - prevent a KeyError exception (#45169)

Ensure ports to integers after allowing the key 'Targets' to be available in params

ryansb committed:
Fix ec2_group for numbered protocols (GRE) (#42765)

* Fix spurious `changed=True` when int is passed as tag

* Fix for all AWS module using compare_aws_tags

* Handle improperly stringified protocols and allow inconsistency between None/-1 on non-tcp protocols

* Add integration test that reproduces the same bug

* Return false if the comparsison is not equal

tedder committed:
docfix, python-netaddr should be netaddr (#43372)

"python-netaddr" is an OS package, but "netaddr" is the pypi package needed in python. Suggesting OS packages for python libs seems in bad form. I like the syntax "python's netaddr" to explain what package manager would have it.

emailbob committed:
Update documentation for gcp_compute (#43791)

Added examples on how to use "keyed_groups", "hostnames", and "compose"

The compose example shows how to set the ansible_host var for a host to either the public or private ip. This is necessary when you set your hostname by name instead of ip

chalupaa committed:
[aws] lambda_policy fix update when principal is an account number (#44871)

Fix KeyError on update when principal is an account number

ianw committed:
docs: Clarify include_task v import_tasks with conditionals (#43856)

Correct and clarify "set_fact" example, expanding on what is happening
in the easy-to-get-wrong import mode. Add some additional links to "group_by" documentation and the main import/include discussion.

Closes: #31596

rlondner committed:
Pure Storage doc fix (#45197)

* Pure Storage doc fix

Fixed typos + aligned formatting between FlashArray and FlashBlade sections

* double, not triple, backticks to quote code

mattclay committed:
Fixes for ansible-test sanity import test. (#45249)

* Fix import test on Python 3.7.
* Fix path processing in import sanity test.

mattclay committed:
Remove obsolete default container.

caphrim007 committed:
Updates documentation formatting and dependencies (#45252)

The documentation formatting was a bit incorrect and the dependencies
mentioned were out of date. This fixes that.

yungezz committed:
add publish url in webapp facts output (#45131)

* add publish url in webapp facts output

* update name

* fix test

bcoca committed:
clarify free strategy (#44645)

* clarify free strategy
* wordsmithing
* shorten line per Shippable

Akasurde committed:
VMware: Handle systemError in host_config_manager (#45263)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

gdpak committed:
iosxr dci failure on devel/2.7 (#45268)

trishnaguha committed:
fix python26 nxos terminal plugin bug (#45271)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

Commit:
Add single quotes around package name (#45152)

<!--- Your description here -->
`pip install ansible[azure]` results in `zsh: no matches found: ansible[azure]` at least in my computer (zsh on Ubuntu). I don't know if it is the case with all shells, but at least for me it is. Since square brackets `[]` are special characters in bash, I propose adding single quotes to make sure that package name is always interpreted as literal. The same error is also resolvable by setting noglob.

bcoca committed:
prefer ansible_facts namespace and dict notation (#44980)

* prefer ansible_facts namespace and dict notation

* fixed mistranslation

remyleone committed:
Select image facts by region (#44965)

bcoca committed:
correct and clarify deprecation (#45234)

* correct and clarify deprecation

maxamillion committed:
fix version number because this was backported

Signed-off-by: Adam Miller <admiller@redhat.com>

ganeshrn committed:
Fix diff_ignore_lines option issue for candidate configuration (#45201)

* Fix diff_ignore_lines option issue for candidate configuration

*  diff_ignore_lines option is to handle the running config fetch from
   remote host and ignore the lines that are auto updated eg: commit time and date
*  This option should not be used while processing candidate (input) configuration

* Fix review comment

ryansb committed:
[aws] ec2_group multi-account and peered VPC bugfix (#45296)

* Add tests to replicate bug #44788 

* Handle when userId is same account due to in-account peering

* Module defaults for main.yml

* Turn off VPC peering tests in CI

wimnat committed:
Fix health check configurations being ignored - #43244 (#43247)

* Fix health check configurations being ignored - #43244

* Ensure health_check_protocol is not None before check

* Simplify long indented if statement

roadmapper committed:
[aws] ec2_metadata_facts URL encode the metadata URL to avoid errors (#43394)

- If a field in the URL has a space it would result in a 400 without making it URL safe
- Fixes #42371 and #43378

Commit:
Adding maintainer for Redfish modules (#45299)

ianw committed:
plugins/callbacks : fix some key documentation (#45267)

These are supposed to be a dictionary with section/key entries;
correct a few typos.

andrewmcgilvray committed:
check for change of role arn in ecs task definition (#44942)

* check role arn for ecs task definition

If the task role in a ECS task definition changes ansible should create a new revsion of the task definition.

s-hertel committed:
Fix backwards compatibility without providing prefix since boto3 does more parameter validation than boto (#45308)

Use .get() instead of assuming a rule has a Transitions or Filter key (both of which are optional)

caphrim007 committed:
Corrects documentation for parameters and examples (#45311)

Documentation for the bigiq modules was a little out of date.
This patch corrects the examples and parameters.

kobayashi committed:
Delete parameters created with init (#42777)

Akasurde committed:
VMware: update vmware_local_role_manager with action (#44566)

With this fix user can add, remove and set privileges to
an existing role with privileges.

Fixes: #44391

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde commented:
@vmwjoseph Could you please add changes suggested with review comments, so that we can take things forward ? Thanks.

MikeKlebolt committed:
[New Module] vmware_guest_tools_upgrade (#42698)

New module to manage VMware guest tools upgrade.

Commit:
consolidate dnszone tests (#45328)

Commit:
consolidate availabilityset tests (#45327)

Im0 committed:
pyvcloud requirement documented (#44347)

Commit:
consolidating dns recordset tests (#45331)

Nothing4You committed:
Correct version_added for strategy host_pinned (#45333)

rmin committed:
VMware: New module vmware_guest_vnc (#36282)

trishnaguha committed:
Add ambiguous command check as the error message is not persistent on nexus devices (#45337)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

Akasurde committed:
VMware: Refactor disc logic (#39285)

* Refactoring related to network device
* Assign unique random temporary key while creating SCSI or/and IDE controller devices
* Add testcase for this change

Fixes: #38679

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

trishnaguha committed:
cli_config module doc update (#45345)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

pieteravonts committed:
VMware: Rewrite get_resource_pool method for correct resource_pool selection (#39792)

* rewrite get_resource_pool method for correct resource_pool selection
* only keep name if path is given for cluster, esxi_hostname or resource_pool
* Revert "only keep name if path is given for cluster, esxi_hostname or resource_pool"
* This reverts commit 50293ec763c024b0eaceac5d775ccc0ad3ff8bd7.
* if the name argument contains a path, only use the last part for matching
* remove path from cluster argument in tests
* remove find_objs in favour of reusing find_obj with an extra folder argument
* fix find_obj ignoring first if name is not given

ryansb committed:
[aws] Remove deprecated `ec2_vpc_dhcp_options` alias of ec2_vpc_dhcp_option (#45321)

ryansb committed:
[aws] Remove deprecated `ec2_vpc_dhcp_options_facts` alias of ec2_vpc_dhcp_option_facts (#45322)

kryptsi committed:
VMware: new plugin for setting an object's role/permission (#39988)

Commit:
rewrite of the developer guide, part 1 (#45179)

* rewrite of the developer guide, part 1

sivel committed:
Ensure loop with delegate_to can short circuit the same as without delegate_to. Fixes #45189 (#45231)

oolongbrothers committed:
Add integration tests for flatpak module

PR #42169

sivel committed:
Don't require requests in hacking/report.py (#45350)

* Don't require requests in hacking/report.py

* move ansible import

morgajel committed:
not sure if typo or pun. (#45358)

Fixing header to be SNapshot
+label: docsite_pr

Meecr0b committed:
tower_credential: expect ssh_key_data to be a string instead of path (#45158)

* expect ssh_key_data to be a string instead of path

ssh_key_data should be a string filled with the private key
the old behavior can be archived with a lookup

Fixes #45119

* clarifies ssh_key_data description, adds newline

oolongbrothers committed:
Add integration tests for flatpak_remote module

PR #42315

jctanner committed:
vmware_guest test - Use the cluster name instead of the path (#45368)

mattclay committed:
Update default-test-container to 1.1.0.

willthames committed:
Use a sensible default for k8s merge_type (#45284)

* Use a sensible default for k8s merge_type

The sensible default for merge_type is `['strategic-merge', 'merge'].
However, we can't make this the default default, as we need to support
users who are using openshift 0.6.0, where the merge_type parameter is
unsupported.

* Refactor k8s test suite for merge_type tests

Allow tests with pre-merge-type openshift and post-merge-type
openshift.

kbreit committed:
meraki_mr_l3_firewall - Integration test fix (#45366)

* Fix integration test as wrong module was called

* Create block section for failures on integration test for mr_l3_firewall integration test

mattclay committed:
Fix tower_project integration test.

Changes required due to https://github.com/ansible/ansible/pull/45158

mattclay committed:
Update default container to version 1.2.0.

kbreit committed:
meraki_network - Fix enabling my.meraki.com (#45352)

* Fix enabling my.meraki.com
- A parameter of "no" wouldn't be picked up
- Check for False in addition to whether it exists

* Clean up disable_my_meraki parameter check

Commit:
added tags to webapp documentation (#45375)

sandeepmendiratta committed:
Update copy.py (#45393)

<!--- Your description here -->
Removed additional  apostrophe in dest:
+label: docsite_pr

stravassac committed:
VMware: more detail error on bad http code return for vmware_guest_file_operation.py (#41897)

Fixes: #41721

Commit:
added missing facts in container instance (#45380)

* returning tags in container instance facts

* small fix

Commit:
Fixes in container registry documentation (#45390)

* adding tags to container registry

* fixed tags

albers committed:
VMware: honor "wait_for_ip_address" when powering on a VM (#45226)

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
VMware: Fix timeout detection in vmware_deploy_ovf module (#45159)

Signed-off-by: Harald Albers <github@albersweb.de>

Akasurde committed:
VMware: use UUID to find templates and VMs (#45208)

This fix adds functionality to find the templates or VMs for cloning
operations.

Fixes: #45081

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
fixing broken sql server tags (#45378)

rwaweber committed:
[linode] add lwm as maintainer, rm me (#45406)

Add lwm, an active user of the linode module to the maintainer list.
Removes myself from the linode maintainer list as I no longer use the Linode module in an occupational capacity and unfortunately don't foresee having much time to devote to cultivating the health of the module(s) in the coming future.

Akasurde committed:
VMware: skip customization if no type is specified (#44510)

VM customization is not necessary if user is not specifying IP address
and netmask. This fix adds exception for type 'DHCP' which is set by default
if no networking details are provided.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
VMware: exclude start_connected from host customization check (#40990)

kbreit committed:
Meraki scenario guide (#41827)

* Initial commit for the Meraki scenario guide

* Added Meraki guide to indexes, fixed an error

* Added common parameters to scenario guide

* Add additional information for first draft
- Added very common parameters everyone uses.
- Documented common format for returned data from Meraki.
- High level explanation of error handling.

* Fix .rst formatting error

* Added section about handling returned data. More to come.

* Small formatting changes

imcdona committed:
Add Sangoma linux (#45430)

Fixes https://github.com/ansible/ansible/issues/45300

jdufresne committed:
Prefer io.BytesIO over six; available on all supported Pythons (#45388)

On all supported Pythons, the io.BytesIO is always a stream
implementation using an in-memory bytes buffer. Makes code slightly more
forward compatible by reducing use of the six module.

mattclay committed:
Reduce cost of Azure DB tests.

fabianvf committed:
Set defaults from params after loading files, allowing params to override (#44142)

* Set defaults from params after loading files, allowing params to override

* cleanup, add some comments

sivel committed:
Don't pass file_name to DataLoader.load in script inventory plugin (#45428)

* Don't pass file_name to DataLoader.load in script inventory plugin. Fixes #34164

* Add changelog fragment

Alexhha committed:
Update postgresql_user.py (#45146)

* Update postgresql_user.py

md5sum generates output with hypen. Something like the following
```
$ echo "md5$(echo -n 'verysecretpasswordJOE' | md5sum)"
md5d011966da94d776cf59bf6dbde240e5d  -
```

We need to remove hyphen from the output. Also the command by itself is incorrect

```
echo "md5$(echo -n 'verysecretpasswordJOE' | md5sum")
```

double quotes must be after right parenthesis

```
echo "md5$(echo -n 'verysecretpasswordJOE' | md5sum)"
```

+label: docsite_pr

* Update postgresql_user.py

Add curly braces

sivel committed:
2.8 Core Deprecation Removal (#45232)

* Remove deprecated ansible.vars.unsafe_proxy. Fixes #45040

* Remove deprecated validate_md5 alias from fetch module. Fixes #45039

* Remove deprecated private arg from import/include_role. Fixes #45038

* All include deprecations bumped to 2.12. Fixes #45037

* Add changelog for deprecated removals

seuf committed:
Grafana datasource module : Use url_argument_spec from urls (#40402)

* Added client_cert and client_key modules parem

* Use url_argument_spec to init module arguments

* Do not compare version and readOnly

* Convert HTTPResponse content to text before json.loads

* Added password in secureJsonPayload when datasource type is postgres

orthanc committed:
Add missing s3 permissions for s3 module testing. (#43243)

jborean93 committed:
reordered windows test groups for better efficiency (#45445)

jborean93 committed:
win_say - fix up syntax and test issues (#45450)

jborean93 committed:
fix Ansible.ModuleUtils.FileUtil to respect ErrorAction if running in a try/catch (#45451)

Commit:
adding tags support to postgresql (#45339)

jborean93 committed:
windows testing - changes to accomodate future behaviour changes (#45464)

jborean93 committed:
win_group_membership - fix random issue with CI on 2012 R2 (#45462)

MikeKlebolt committed:
Fixes authentication error when ansible_user is a local account and computer is already joined to domain (#44270)

* Fixes authentication error when ansible_user is a local account and computer is already joined to domain

* recommended fixes

jborean93 committed:
win_scheduled_task: add deprecation warning for repetition format (#45468)

* win_scheduled_task: add deprecation warning for repetition format

* fixed up sanity issues

jborean93 committed:
Win deprecate 2.8 (#45473)

* Remove deprecated/expired functionality

* win:_msi: Remove the win_msi module

* removed some missed deprecated return options and added porting guide reference

Co-authored-by: dagwieers <dagwieers@users.noreply.github.com>

albers committed:
VMware: Fix wait_for_task backoff behavior (#45429)

Signed-off-by: Harald Albers <github@albersweb.de>

Commit:
fixing facts for mysql server (#45374)

jamescassell committed:
remove unused templar._clean_data() (#42739)

yuwzho committed:
fix the autoscale resource id (#45477)

* fix the autoscale resource id

* Update main.yml

ptux committed:
Promote ptux to a co-maintainer for copy

PR #45466

samdoran committed:
Add documentation about patterns with commas to find module (#45360)

Thanks @samdoran  for the example and clarification!

mscherer committed:
Add documentation for the modify command, by giving a example (#19699)

webknjaz committed:
Add a ticket stub for guiding GitHub newcomers

PR #45497

Commit:
Adding maintainers to Redfish modules

PR #45443

samdoran committed:
Add docs on how to write changelog fragments (#42428)

* Add docs on how to write changelog fragments

* Make recommended changes

* Restore lines and make recommended changes

* Add link to new style changelog

Remove redundant mention of old style changelog

deivid-rodriguez committed:
Fix remote checksums when paths have leading dots (#45287)

* Fix remote checksums with paths have leading dots

* Fix result recorded from the wrong file

* Add changelog fragment

Commit:
Remove more docs build errors (#45364)

* orphans testing pages to avoid not-in-toctree errors

* orphans various pages pending reorg

* adds module_utils and special_vars to main TOC

* uses a glob for scenario_guide TOC

* normalize and Sentence-case headings on community pages, typos

* re-orgs community TOC, adds all pages to toctree

* removes scenario guides index page

* adds style guide to community index

* basic update to style guide

* fix typo that created a new error

* removes not-in-toctree from ignore errors list

* leave removing files for future cleanup task

gtema committed:
update porting guides (#40784)

* update porting guides

With PR #40532 `shade` library was retired and replaced with direct use
of `openstacksdk`. Porting guides and doc about dynamic inventory were
not updated.

mattclay committed:
Avoid duplicate Azure storage account types.

mattclay committed:
Fix ACI unit test on Python 3.7.0.

The previous logic was only needed for pre-release versions of 3.7.

Qalthos committed:
Fixup network connection documentation (#45487)

* Fixup network connection documentation similar to suggestions in #45224

* Another tpyo

sivel committed:
Fix logic to not re-download existing files when force=no (#45495)

* Fix logic to not re-download existing files when force=no. Fixes #45491

* Reduce logic complexity

s-hertel committed:
[AWS] cloudfront_distribution - fix method name from 'validate_distribution_id_from_caller_reference' to 'validate_distribution_from_caller_reference' and set distribution_id to the distribution's key 'Id' (#45498)

FriedCircuits committed:
Fix python 3 compatibility for network id lookup (#45467)

Due to a change in python 3 filters return filter instead of list. This breaks the network name to id lookup for the docker_swarm_module. By wrapping it in list it ensures it is a list when returned and the id is extracted.

nkakouros committed:
Update implicit_localhost.rst (#45455)

Adds a clarification about `inventory_dir/file` magic variables when using implicit localhost as per #45385 . +label: docsite_pr

flatline-studios committed:
Clarify the creates and removes options (#45485)

* Clarify the creates and removes options

The description for the creates and removes options were worded poorly. This change improves readability and is easier to understand.

 Double negatives are BAD

+label: docsite_pr

* Fix trailing whitespace

bcoca committed:
updated fbsd install instructions (#45309)

Commit:
lineinfile note belongs in changelog for 2.6, not 2.7 (#45517)

Commit:
Update the root URI to match the DMTF specification (#45295)

mscherer committed:
Fix useless comparaison between 2 constants (#45524)

Found with https://lgtm.com

mattclay committed:
Update RHEL and FreeBSD versions used for tests:

- RHEL 7.4 -> 7.5
- FreeBSD 11.1 -> 11.2

gdpak committed:
asa_config/ios_config: diff strict does not work with multiple parents (#45150)

* multiple parents issues in diff

* Integration tests for missing functionality

* add testcase for other platforms. vnxos does not support qos so need to find a command chain on v-nxos for multiple parets. junos uses on-device diff so should not need this.

* Fix for issue when any candidate parent did not meet the exact line in running-config

* DCI runs eos_config without become flag

Commit:
fixed tag support in autoscale facts (#45533)

* fixed tag support in autoscale facts

* forgot to add tags to arg spec

buzz3791 committed:
Update keycloak_client.py (#45505)

Slightly improve wording about lack of input sanity checking on API calls.

+label: docsite_pr

Commit:
extended vmss facts (#45535)

mkrizek committed:
BOTMETA: fix notified key (#45419)

Commit:
Azure load balancer crash fix (#45546)

Ghilli3 committed:
Initial commit of fmgr_sys_proxy.py (#45518)

Qalthos committed:
eos: Don't error on VLAN missing warning (#45433)

abirami-n committed:
Fixes handling of terminal_initial_prompt on dellos6 and dellos9 devices (#45273)

* Fix terminal initial prompt on dellos6 and dellos9

* Fix_pylint

thetuxkeeper committed:
fixing bytes <-> str regexp issue in cloudengine (#45285)

dagwieers committed:
Fix integration tests (#43847)

I doubt this is the correct test, but until we have this fixed/improved
at least this works fine.

Commit:
Lenovo cli bugfixes (#45278)

* To change all CLIs present in the code to latest CNOS CLI. One bug of executing with Lenovo Jakku switch is also getting fixed.

* To change the file names as the commands have changed from display to show

* To change to new CLI in test_cnos_comand.py as well

Qalthos committed:
Clean up after cnos tests (#45564)

Nothing4You committed:
Make it clear that recurse only applies to state=directory (#40222)

This would otherwise cause confusion as it could be interpreted as only changing attributes for directories (and not files inside).

+label: docsite_pr

skymoore committed:
Update README.md (#37272)

Update Arch packaging README.md with link and package mgr options

Evertvz committed:
Added a note regarding the delegate_facts option. (#25257)

* Added a note regarding the delegate_facts option with a link to documentation

bcoca committed:
moving from with_ff to just ff (#44847)

FloMiau committed:
Change "Defaulting Undefined Variables" (#41379)

Rewording the text for the second parameter of default because the original text was misunderstandable. +label: docsite_pr

caphrim007 committed:
Fixes bugs in the bigip_software modules (#45577)

This patch makes hotfix code work, where before, only base images
were working

Commit:
adds license details to dev guide pages (#45574)

* adds license details to dev guide pages

* fixes capitalization error

dagwieers committed:
Docs: Make docsite rebuilds smarter/faster (#45062)

* Make the following scripts  idempotent so that we only have to rebuild changed docs, not all docs:

  * plugin_formatter
  * generate_man
  * dump_keywords.py
  * dump_config.py
  * testing_formatter.sh

fholzer committed:
Clarify how java_cert checks keystore certificates (#26567)

JonTheNiceGuy committed:
Show multi-line messages in debug documentation (#23000)

* Show multi-line messages in debug documentation

* Add two-line entry
* Indicate that jinja2 filters are not applied here

* Fix to incorrect documentation change

pilou- committed:
FAQ: fix a typo, add link to 'vars' lookup (#42412)

* FAQ: add link to vars plugin doc

* Typo (s/Techinically/Technically/)

mnecas committed:
Ovirt add rename functionality (#44951)

* add renaming of ovirt
* add renaming func to ovirt cluster
* ovirt update examples in docs

samdoran committed:
Improve code stability is checksum checking

- use context manager for dealing with the checksum file
- use loop that can tolerate zero, one, or more items return rather than the previous expression which would break if anything other than exactly one item was returned

jborean93 committed:
win_script: add support for become and centralise exec wrapper builder (#45582)

* win_script: add support for become and centralise exec wrapper builder

* satisfying the pep8 gods

* do not scan for module dependencies when running as a script

Qalthos committed:
Exclude the whole ansible_failed_task block (#45587)

It has lots of FieldAttributes that won't serialize.
Also the connection doesn't care.

nkakouros committed:
Update fileglob.py (#45558)

* Update fileglob.py

Closes #45557

* Update fileglob.py

kustodian committed:
Add an error message if a pkg cannot be removed, fixes #35672 (#40723)

Commit:
adding sql firewall rule facts (#45059)

Qalthos committed:
network command modules cleanup (#45242)

* Standardize _command modules

* Centralize command transformation process

* Fix check mode error test

annikulin committed:
Store Authorization header inside HttpApi connection plugin (#45598)

mscherer committed:
Remove unused import iControlUnexpectedHTTPError (#45611)

Since that's unused in the code, no need to import it.
This got flagged by linter

dagwieers committed:
Fix link and toctree (#45595)

caphrim007 committed:
Fixes lgtm alert (#45619)

Alert was for an include that was not used.
https://lgtm.com/projects/g/ansible/ansible/latest/files/lib/ansible/modules/network/f5/_bigip_facts.py?sort=name&dir=ASC&mode=heatmap&showExcluded=false#L98

dagwieers committed:
Improve the local toctree (and title) (#45590)

Make toctree for Windows dev guide local, remove 'Ansible' from the title, since
this documentation is about Ansible.

pilou- committed:
hashi_vault lookup: add integration tests (#44814)

* hashi_vault lookup: add integration tests

* hashi_vault lookup tests: use ansible-ci-files

kbreit committed:
meraki - Remove YAML formatting on JSON code block (#45526)

* Remove YAML formatting on JSON code block

shinmog committed:
updating modules/network/panos reviewers (#45622)

vibhavsinha committed:
fix gce_backend_service examples for healthchecks (#31134)

gce_backend_service module expects healthchecks to be an array of string. The previous example incorrectly mentioned healthchecks as an array of objects each containing a `name` key.

sivel committed:
Support empty files with piped transfer_method. Fixes #45426 (#45618)

caphrim007 committed:
Fixes multiple lgtm issues (#45629)

Most of the issues were just inclusions of code that were not used.
This cleans those up. Other alerts were semi-false-positives for now

Commit:
Merge branch 'vmkernel' of github.com:vmwjoseph/ansible into vmkernel

bcoca committed:
avoid empty groups in json output (#45621)

they get confused as hosts by script plugin

fixes #45601

Commit:
Address initial review feedback

Commit:
Fix alias

vmwjoseph commented:
I've made a mess of this trying to rebase, I'll reopen a new pull request with the feedback above and retesting.
